### ISSUE TITLE
Allow EEGclassifier in cropped to work with gridsearch cv or pipeline

### ISSUE THREAD

**Issue by bruAristimunha** [braindecode/braindecode#438]
Hi @robintibor and @agramfort!

At the moment, as discussed in #422, the EEGClassifier with cropped option doesn't work when we put the EEClassifier inside a Gridcvsearch or a Pipeline:

Robin had suggested two options to solve this problem:

a) use manual cross-validation
b) simulate cropped decoding by using trialwise decoding and adding a global average pooling layer at the end of your model

I would like to implement the second option to solve directly in the core. I would like to know if, on your side, do you see any blockages or considerations in mind?
            
_Originally posted by @robintibor in https://github.com/braindecode/braindecode/issues/422#issuecomment-1285651484_

**Comment by robintibor** [comment]
So do you mean adding this as an additional option? or how do you mean solve directly in the core?  Like you want to replace the current style of cropped decoding or you want to add some new option alongside?

**Comment by bruAristimunha** [comment]
Hey Robin!

I hope you're doing well. I would like your thoughts on something. I've been figuring out how to make the cropped option work with the cross-validator, and I'm running into some issues.

Maybe we could implement your suggestion from the PR and add the pooling layer, but if that's not the right solution, I'm open to other ideas. My main priority is ensuring the eegclassifier object can be cloned and used for a gridsearch.

Right now, we are having this type of error:
![image](https://user-images.githubusercontent.com/42702466/223559955-3e4c07e7-d76e-4bf3-96a6-08273c522dba.png)

It seems to be related to the function predict_with_window_inds_and_y, which you can find here:
https://github.com/braindecode/braindecode/blob/c22f0b1f113a4523abc8bce3663c48850ed65455/braindecode/classifier.py#L178

**Comment by agramfort** [comment]
option b) seems the way to go for me

**Comment by robintibor** [comment]
Yeah I also think adding average pooling and not using cropped mode could be a simple solution for this use case, do you think this is enough info for you already? otherwise we can also zoom on Friday if you need and have time

**Comment by robintibor** [comment]
Yeah in two weeks could be a good time to discuss that
